queue/priority_queue.go

```diff
@@ -0,0 +1,65 @@
+package queue
+
+import (
+	"container/heap"
+)
+
+type PriorityQueueItem interface {
+	Less(other PriorityQueueItem) bool
+}
+
+type priorityQueueRepr []PriorityQueueItem
```

Comment:
What's "repr" here mean?

Reply:
It meant "representation" but removed it to be less confusing.